docs/docs/reach-router-and-gatsby.md

```diff
@@ -20,6 +21,46 @@ With Gatsby, you will mostly be using the `<Link />` component provided by the `
 
 Besides using the [`<Link />` API](https://www.gatsbyjs.org/docs/gatsby-link/) for linking between pages Gatsby generates, you can define your own client-side routes. See the [client only paths example](https://github.com/gatsbyjs/gatsby/tree/master/examples/client-only-paths) on how to use `<Router />` from `@reach/router` to make client routes work seamlessly together with your server routes.
 
+## Scroll Restoration
+
+Gatsby will handle scroll restoration for you in most cases. However, when you render containers that have their own scroll values, those scroll positions are typically lost between page transitions. To solve that, users can use the (deprecated) `ScrollContainer` component or `useScrollRestoration` hook in their code to tell Gatsby about scroll containers that we should track and restore.
```

Comment:
I'd list the recommended `useScrollRestoration` before the deprecated fallback option. Thank you for adding these, it's so helpful! I was just crawling through code trying to figure out how it works for CSR a11y improvements.

Reply:
Awesome! Will do

Reply:
Oooo, so I assumed this was an a11y improvement but I didn't know for sure. Should we add a reference to this from any a11y documentation?